### ISSUE TITLE
If screen is at 9:16 ratio, GUI is extremely elongated

### ISSUE THREAD

**Issue by feilen** [mabrowning/minecrift#140]
If the window minecraft is running in is at 1080x1920 resolution, then the GUI inside will be extremely elongated. Ordinarily this wouldn't crop up as a problem, but in Linux (with the OVR Linux distortion caps) you're supposed to leave the rift screen unrotated and let the SDK take care of it for maximum performance. 

To get around this we need a way to force the aspect ratio of the GUI, ideally.

**Comment by feilen** [comment]
What does IStereoProvider.isGUIOrtho() refer to? I don't see any comments even mentioning it but it's used a lot.

**Comment by StellaArtois** [comment]
Basically ortho is a fullscreen, non 3d GUI. So for stereo rendering, isGuiOrtho will be false.

**Comment by feilen** [comment]
Ah okay. Doesn't really seem so be one central place I can set the gui
ratio, so it'll take a bit.

**Comment by StellaArtois** [comment]
That's something that's been on my todo list for sometime (one central
place for GUI resolution / aspect)! Exactly what's needed :-)

**Comment by feilen** [comment]
I'll give it a shot! It's pretty important to Linux performance, as I think
with XRandR it has some slow method of rotation. Not ideal.

**Comment by StellaArtois** [comment]
This may also fix an issue with GUI scaling in Mono non-stereo mode with
a FSAA scale factors that are non power of 2.

**Comment by feilen** [comment]
I'll give it a look. Right now, my laptop's SSD is dying to I have to move
a few things around it. Should be good soon. I'll keep thinking of things I
want to implement, but for now you've got quite a large handful of issues
you could probably close (ones older than six months or that won't get
fixed or that have nothing to do with Minecrift)

**Comment by StellaArtois** [comment]
Yes, agreed. I need to do some housekeeping!

**Comment by feilen** [comment]
Oh! Also, should probably set the 'default branch' in github settings to
the 1-8-1 one. Same with port-to-etc on JRift.

**Comment by feilen** [comment]
I can't get my reobfusicated changes to work, but they launch fine out of Eclipse. 
Out of curiousity, could you give me your .classpath and .project files so I can make sure I haven't missed something obvious? I spent a good hour fiddling with the classpath to get it in the right order to even run it in the editor, so some baseline would be appreciated.

**Comment by StellaArtois** [comment]
I use Intelij, but point the editor at the lib/1.8.1 dir in the root of the
git repo for my class path (add all files). This dir is generated when
running ./install.sh. The natives are also in there. I point Intelij at the
Start class for startup.

This all needs to be a maven or gradle project really. Would love to use
the forge project as a base for Minecrift in the future -as it has all this
sorted.

**Comment by feilen** [comment]
Well, I got it reobusicating perfectly fine before. When you changed out
the line endings decided to re-do my mcpxxx from scratch, and now it
appears to be having issues. I like Eclipse personally.

It's already pointed to that directory I think, but if I can run straight
off that it might be simpler. I'm just going off the base file in
eclipse/Client.

**Comment by StellaArtois** [comment]
Yup, I’ve definitely broken something. I’ll have a look into it (again!)

**Comment by feilen** [comment]
Haha. Have you tried reobfuscation? Did it work?

**Comment by StellaArtois** [comment]
No, not yet! Testing isn’t my forte, apparently!

**Comment by virtuoushub** [comment]
@StellaArtois, At one point long ago I had a branch that [used Maven](https://github.com/mabrowning/minecrift/issues/140#issuecomment-72501870) as the build system. I did not get it configured quite how I liked and I never finished merging it back in.

We had previously talked about Forge compatibility. I am not sure if it will help to resolve this issue ( #140 ) but I would be happy to look into Forge compat. if no one else is working on it.

**Comment by StellaArtois** [comment]
Hi Peter, if you would like to look at Forge compatibility that would be
most appreciated! As far as I know nobody is currently actively looking at
it.

**Comment by StellaArtois** [comment]
Zach, EntityRenderer should be fixed now. Can get everything out a-fresh, it decompiles, builds, reobfuscates AFAIKT.=

**Comment by StellaArtois** [comment]
Just pushed some more changes to the repo - and now - the patches are
borked again. Definitely something up, will investigate further.
